ch_pipeline/hfb/flagging.py

```diff
@@ -348,6 +345,66 @@ def process(self, rfimaskbitmap):
         return out
 
 
+class RFIMaskReduceBeamNS(task.SingleTask):
+    """Create RFIMask from HFBDirectionalRFIMaskBitmap.
+
+    This task takes an HFBDirectionalRFIMaskBitmap)freq, beam_ns, time), selects the RFI mask corresponding to a specified significance value used in the detection and subfrequency threshold, then reduce the 'beam_ns' axis to create a RFIMask container (freq, time).
+
+    Attributes
+    ----------
+    sigma : int
+        Specify the value of significance used in RFI detection to create a RFI mask.
+        Must match one of the sigma keys used when the HFBDirectionalRFIMaskBitmap was
+        created. Default is 5.
+    subfreq_threshold : int
+        Subfrequency threshold for decoding the bitmap. A sample is flagged if
+        the number of HFB subfrequency channels detecting RFI >= this threshold.
+        Default is 2.
+    beam_ns_threshold : int
+        This number determines the minimum number of detected RFI events along the beam_ns axis required for a data point
+        to be included in the reduced mask. Default is 1.
+    """
+
+    sigma = config.Property(proptype=int, default=5)
+    subfreq_threshold = config.Property(proptype=int, default=2)
+    beam_ns_threshold = config.Property(proptype=int, default=1)
+
+    def process(self, rfimaskbitmap):
+        """Produce a RFI mask.
+
+        Parameters
+        ----------
+        rfimaskbitmap : containers.HFBDirectionalRFIMaskBitmap
+            beam_ns-specific RFI mask with axes (freq, beam_ns, time).
+
+        Returns
+        -------
+        out : draco.containers.RFIMask(freq, time)
+            Non beam_ns-specific RFI mask with axes (freq, time).
+
+        """
+        # Extract mask/frac and axes data
+        mask = rfimaskbitmap.get_mask(self.sigma, self.subfreq_threshold)[:]
+        beam_ns_axis = list(rfimaskbitmap.subfreq_rfi.attrs["axis"]).index("beam_ns")
+        freq = rfimaskbitmap.freq[:]
+        time = rfimaskbitmap.time[:]
+
+        # Apply reduction condition
+        reduced_mask = np.sum(mask, axis=beam_ns_axis) >= self.beam_ns_threshold
+
+        # Create an output container
+        output = RFIMask(freq=freq, time=time)
+
+        # The output RFI mask is not frequency distributed
+        arr = reduced_mask
+        arrdist = mpiarray.MPIArray.wrap(arr, axis=0)
+        final_mask = arrdist.allgather()
+        output.mask[:] = final_mask
```

Comment:
```suggestion
        output.mask[:] = mpiarray.MPIArray.wrap(reduced_mask, axis=0).allgather()
```
This can just be reduced to a single line, which in some way is more readable because there aren't intermediate variables.